src/date-range-input/date-range-input.tsx

```diff
@@ -316,6 +316,16 @@ export const DateRangeInput = ({
             return;
         }
 
+        const isInvalidRange = dayjs(val).isBefore(selectedStart, "day");
+
+        // if date range is invalid, set selected value as start and reselect end
+        if (isInvalidRange) {
+            actions.changeStart(val);
+            calendarRef.current.setCalendarDate(val);
+            actions.reselectEnd();
+            return;
+        }
```

Comment:
this is breaking keyboard input: upon typing end date < start date, it updates the start date and either focuses on the wrong input or does not reset the end date at all.